ethcore/builtin/src/lib.rs

```diff
@@ -215,61 +217,58 @@ impl Builtin {
 	}
 }
 
-impl From<ethjson::spec::Builtin> for Builtin {
-	fn from(b: ethjson::spec::Builtin) -> Self {
+impl TryFrom<ethjson::spec::Builtin> for Builtin {
+	type Error = Error;
+
+	fn try_from(b: ethjson::spec::Builtin) -> Result<Self, Self::Error> {
 		let pricer: Box<dyn Pricer> = match b.pricing {
-			ethjson::spec::Pricing::Blake2F { gas_per_round } => {
-				Box::new(gas_per_round)
-			},
-			ethjson::spec::Pricing::Linear(linear) => {
-				Box::new(Linear {
-					base: linear.base,
-					word: linear.word,
-				})
-			}
-			ethjson::spec::Pricing::Modexp(exp) => {
-				Box::new(ModexpPricer {
-					divisor: if exp.divisor == 0 {
-						warn!(target: "builtin", "Zero modexp divisor specified. Falling back to default.");
-						10
-					} else {
-						exp.divisor
-					}
-				})
-			}
-			ethjson::spec::Pricing::AltBn128Pairing(pricer) => {
-				Box::new(AltBn128PairingPricer {
-					price: AltBn128PairingPrice {
-						base: pricer.base,
-						pair: pricer.pair,
-					},
-					eip1108_transition_at: b.eip1108_transition.map_or(u64::max_value(), Into::into),
-					eip1108_transition_price: AltBn128PairingPrice {
-						base: pricer.eip1108_transition_base,
-						pair: pricer.eip1108_transition_pair,
-					},
-				})
-			}
+			ethjson::spec::Pricing::Blake2F { gas_per_round } => Box::new(gas_per_round),
```

Comment:
One suggestion to cut down on the noise a little bit would be to add a local `use ethjson::spec::Pricing::*;` right before the `match`. Entirely optional though! :)

Reply:
I actually ended up with a bunch of formatting changes here that were unnecessary so I reverted those.